avocado/core/loader.py

```diff
@@ -44,6 +44,52 @@
 ALL = True
 
 
+def filter_test_tags(test_suite, filter_by_tags, ignore_empty=False):
+    """
+    Filter the existing (unfiltered) test suite based on tags
+
+    The filtering mechanism is limited to INSTRUMENTED tests, that is it does
+    not apply to SIMPLE tests or any other test type.  It mean non-INSTRUMENTED
+    tests are *always* included.
```

Comment:
This is true in the sense that no other loader type currently sets `tags`.  We don't have documented instructions for other test types and so on.

IMHO, we should assume that this is only supported for INSTRUMENTED tests at this point, and announce it loudly and explicitly.  If we try it out, and document usage with other test types, than we can raise this "soft" limitation.